server/src/metadata.rs

```diff
@@ -0,0 +1,1310 @@
+        let schema = Schema::new(vec![
+            Field::new("tag", DataType::UInt8, false),
+            Field::new("timestamp", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("value", ArrowValue::DATA_TYPE, false),
+        ]);
+
+        let result = create_simple_model_table_metadata(schema);
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_cannot_create_model_table_metadata_with_no_fields() {
+        let schema = Schema::new(vec![
+            Field::new("tag", DataType::Utf8, false),
+            Field::new("timestamp", ArrowTimestamp::DATA_TYPE, false),
+        ]);
+
+        let result = create_simple_model_table_metadata(schema);
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn test_cannot_create_model_table_metadata_with_invalid_field_type() {
+        let schema = Schema::new(vec![
+            Field::new("tag", DataType::Utf8, false),
+            Field::new("timestamp", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("value", DataType::UInt8, false),
+        ]);
+
+        let result = create_simple_model_table_metadata(schema);
+        assert!(result.is_err());
+    }
+
+    /// Return metadata for a model table with one tag column and the timestamp column at index 1.
+    fn create_simple_model_table_metadata(
+        schema: Schema,
+    ) -> Result<ModelTableMetadata, ModelarDBError> {
+        ModelTableMetadata::try_new("table_name".to_owned(), schema, vec![0], 1)
+    }
+
+    #[test]
+    fn test_cannot_create_model_table_metadata_with_duplicate_tag_indices() {
+        let schema = get_model_table_schema();
+        let result = ModelTableMetadata::try_new("table_name".to_owned(), schema, vec![0, 1, 1], 3);
+
+        assert!(result.is_err());
+    }
+
+    fn get_model_table_schema() -> Schema {
+        Schema::new(vec![
+            Field::new("location", DataType::Utf8, false),
+            Field::new("install_year", DataType::Utf8, false),
+            Field::new("model", DataType::Utf8, false),
+            Field::new("timestamp", ArrowTimestamp::DATA_TYPE, false),
+            Field::new("power_output", ArrowValue::DATA_TYPE, false),
+            Field::new("wind_speed", ArrowValue::DATA_TYPE, false),
+            Field::new("temperature", ArrowValue::DATA_TYPE, false),
+        ])
+    }
+
+    #[test]
+    fn test_cannot_create_model_table_metadata_with_too_many_fields() {
+        // Create 1025 fields that can be used to initialize a schema.
+        let fields = (0..1025)
+            .map(|i| Field::new(format!("field_{}", i).as_str(), DataType::Float32, false))
+            .collect::<Vec<Field>>();
+
+        let table_name = "table_name".to_owned();
+        let result = ModelTableMetadata::try_new(table_name, Schema::new(fields), vec![0, 1, 2], 3);
+
+        assert!(result.is_err());
+    }
+}
+
+#[cfg(test)]
+/// Module with utility functions that return the metadata needed by unit tests.
+pub mod test_util {
```

Comment:
I do not think it is absolutely necessary but it might be relevant to add a doc comment to test utility functions that are meant for wider use such as the functions in separate test utility modules.

Reply:
Fixed.